### ISSUE TITLE
Background playback

### ISSUE THREAD

**Issue by avently** [ram-on/SkyTube#143]
This issue about https://github.com/ram-on/SkyTube/issues/6
You didn't answer to me so i created new issue.
I added background playback to SkyTube from NewPipe. Most of the code i got from NewPipe and adapted it to SkyTube "reality". I'm not a Java programmer so expect bad practics in code. That's why i'm posting it here as a patch. You can try it and if you'll like it you can add it to your repo. It is not a 100% good solution but it works for me like i want it to work. Hope you'll find time and make this solution better.
P.S. Thanks goes to NewPipe devs.

cd SkyTube
git apply "backgroundplayback.txt location"
[backgroundplayback.txt](https://github.com/ram-on/SkyTube/files/1244707/backgroundplayback.txt)

**Comment by ram-on** [comment]
Hi @avently thanks for your input.  i'll look into it.

**Comment by avently** [comment]
@ram-on is doing nothing with this issue so, guys, feel free to use my fork if you want background playback and swipe controls in SkyTube
https://github.com/avently/skytube

**Comment by ram-on** [comment]
@avently can you update your fork to use the latest https://github.com/ram-on/extractor.git ?  I'd like to try it out after that.

**Comment by avently** [comment]
@ram-on done

**Comment by strk** [comment]
@avently do you publish an .apk for your fork version?

**Comment by avently** [comment]
@strk no. It doesn't have new features from original app because it is a little bit outdated. Maybe in the future background playback will be here too

**Comment by rmenessec** [comment]
I'd really like to see basic support background playback, even if it doesn't have notification controls to begin with. I'm using [NewPipe](https://github.com/TeamNewPipe/NewPipe) in the meantime.

**Comment by Davy49** [comment]
Hi,
The last time I tried using the New Pipe application, it wouldn't allow me to play videos in the background either. As I remember, previously when I had a different version of the app installed, it did in fact allow background playing of videos, so I uninstalled the app and tried a different application ( Sky Tube ), now I'm having the same issue of playing videos in the background with it.
David

**Comment by ram-on** [comment]
Background play is currently partially supported.  If the app is minimized, the video will still continue playing.

**Comment by avently** [comment]
@ram-on in background video stream is used, right? I think video stream should be changed to audio only stream because of the video vs audio size in MBs. 

Also I noticed that bottom navigation is not show up in new versions while video is playing. So its impossible to view virtual buttons (back, home, etc). It's not a big problem so new issue is not needed

**Comment by ram-on** [comment]
@avently To view the bottom navigation bar, just swipe from the very top (to bottom).

**Comment by avently** [comment]
@ram-on its working bad. Because swipe from bottom changes video position while controls are visible or shows a video description. Its ok to hide virtual buttons while video controls are hidden but its not ok to hide them while controls are visible

**Comment by ram-on** [comment]
@avently You need to swipe from the top (i.e. the location where time/date is normally displayed) downwards -- refer to https://i.imgur.com/N0Tlv1B.png

**Comment by avently** [comment]
@ram-on yes, I know about that "workaround". So you think your app should work like this?

**Comment by ram-on** [comment]
@avently That's how it is intended to be.

**Comment by aspieln3r** [comment]
it'd be great if a notification is shown when video is playing in the background

**Comment by Davy49** [comment]
Hi,
Thanks so much for the updated information !
David

**Comment by hamishmb** [comment]
So this can be closed now?
